### PR TITLE
[radio] add a helper of SFD handler

### PR DESCRIPTION
**Pull request by bukepo**
This commit adds a helper of SFD handler to process IEs and security in the SFD handler from ISR.

### CODE REVIEW

**Review by abtink — APPROVED** [review]
Looks good. 👍 
Some smaller suggestions below:

File: examples/platforms/utils/mac_frame.cpp [hunk 1]
```diff
@@ -277,3 +281,100 @@ void otMacFrameSetEnhAckProbingIe(otRadioFrame *aFrame, const uint8_t *aData, ui
...
+
+    otMacFrameProcessTransmitAesCcm(aFrame, &aRadioContext->mExtAddress);
+
+exit:
+    return error;
+}
+
+#if OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
+void otMacFrameUpdateTimeIe(otRadioFrame *aFrame, uint64_t aRadioTime, otRadioContext *aRadioContext)
+{
```
[the remark this answers is not in the document]
Comment (bukepo) [reply examples/platforms/utils/mac_frame.cpp:350]: This PR mainly moves the code. We may refactor it in separate PRs. Thoughts?

[the remark this answers is not in the document]
Comment (zhanglongxia) [reply examples/platforms/utils/mac_frame.cpp:350]: Great. We can move it to a separate PR.

**Review by zhanglongxia — COMMENTED** [review]
LGTM

**Review by zhanglongxia — APPROVED** [review]
LGTM

File: examples/platforms/utils/mac_frame.h [hunk 2]
```diff
@@ -334,6 +334,54 @@ uint8_t otMacFrameGenerateEnhAckProbingIe(uint8_t *aDest, const uint8_t *aIeData
...
+ *
+ * This function may do multiple tasks as follows.
+ *
+ *  - CSL IE will be populated (if present)
+ *  - Time IE will be populated (if present)
+ *  - Tx security will be performed (including assignment of security frame counter and key id if not assigned)
+ *
+ * @param[in,out]   aFrame          The target frame. MUST NOT be `NULL`.
+ * @param[in]       aRadioTime      The radio time when the SFD was at the antenna.
+ * @param[in,out]       aRadioContext   The radio context accessible in ISR.
```
Comment (jwhui) [inline examples/platforms/utils/mac_frame.h:365]:
```suggestion
 * @param[in,out]   aFrame          The target frame. MUST NOT be `NULL`.
 * @param[in]       aRadioTime      The radio time when the SFD was at the antenna.
 * @param[in,out]   aRadioContext   The radio context accessible in ISR.
```

**Review by jwhui — APPROVED** [review]

**Pull request merged**